src/app/games/tafl/tafl.component.ts

```diff
@@ -101,50 +69,54 @@ export class TablutComponent extends RectangularGameComponent<TablutRules,
         }
     }
     private async chooseDestination(x: number, y: number): Promise<MGPValidation> {
-        display(TablutComponent.VERBOSE, 'TablutComponent.chooseDestination');
+        display(this.VERBOSE, 'TaflComponent.chooseDestination');
 
         const chosenPiece: Coord = this.chosen;
         const chosenDestination: Coord = new Coord(x, y);
-        let move: TablutMove;
+        let move: M;
         try {
-            move = new TablutMove(chosenPiece, chosenDestination);
+            move = this.moveGenerator(chosenPiece, chosenDestination);
         } catch (error) {
             return this.cancelMove(error.message);
         }
         this.cancelMove();
         return await this.chooseMove(move, this.rules.node.gameState, null, null);
     }
     public choosePiece(x: number, y: number): MGPValidation {
-        display(TablutComponent.VERBOSE, 'TablutComponent.choosePiece');
+        display(this.VERBOSE, 'TaflComponent.choosePiece');
 
-        if (this.board[y][x] === TablutCase.UNOCCUPIED) {
+        if (this.board[y][x] === TaflPawn.UNOCCUPIED) {
             return this.cancelMove(RulesFailure.MUST_CHOOSE_PLAYER_PIECE());
         }
         if (!this.pieceBelongToCurrentPlayer(x, y)) {
             return this.cancelMove(RulesFailure.CANNOT_CHOOSE_OPPONENT_PIECE());
         }
 
         this.chosen = new Coord(x, y);
-        display(TablutComponent.VERBOSE, 'selected piece = (' + x + ', ' + y + ')');
+        display(this.VERBOSE, 'selected piece = (' + x + ', ' + y + ')');
         return MGPValidation.SUCCESS;
     }
     public pieceBelongToCurrentPlayer(x: number, y: number): boolean {
         // TODO: see that verification is done and refactor this shit
```

Comment:
what about this TODO?

Reply:
Mmh, me semble fait, y avais pas trop une emphase sur l'UT et le coverage à l'époque du vieux tablut, c'était le quatrième quand même, faut pas l'oublier l'enfant d'cul !